src/app/store/Cart/Cart.dispatcher.js

```diff
@@ -123,10 +123,11 @@ export class CartDispatcher {
                 let configurableVariantIndex = 0;
 
                 const { product: variant } = variants.find(
-                    ({ product }, index) => {
+                    ({ product }) => {
+                        if (product === null) return false;
                         const { sku: productSku } = product;
                         const isChosenProduct = productSku === sku;
-                        if (isChosenProduct) configurableVariantIndex = index;
+                        if (!isChosenProduct) configurableVariantIndex++;
```

Comment:
Do not really understood the logic: instead of applying looped index, you are increasing the index? 

I believe this is incorrect, please clarify :)

Reply:
Actually, configurableVariantIndex was set to index of variant which is chosen, but if a product have 5 variants in which 1st, 3rd and 5th Variant is null 
```
Varainats: [ {null},{product}, {null}, {Product}, {null} ]
```
so for our condition to only check if Variant isn't **null**. 
We have filter the Variant List to match the List being passed to Mini-cart and Cart Page.
which turns the upper list to 
```
Varainats: [ {product}, {Product} ]
```
So index 1st, 3rd and 5th are removed from Varaints on Category Page, Product Page, Mini-cart, Cart Page. 
But updateCartData has the Original List so assigning them the index from original list will create problem for Updating Cart Data. 

Line-127 checks if product isn't null then sku is compared and index is set according to condition if its not the same variant added to cart it will increase one index.
Let me know if u still confused.